### DISCUSSION TITLE
DHCP/DNS/Samba server setup

### DISCUSSION THREAD

**Discussion by Angelo325** [cockpit-project/cockpit#18875] [Q&A] [answered]
Hello,

I use the remote administration software "Cockpit" on Debian 11 (Bullseye) and I would like to create a DHCP server, a DNS server and a Samba server with "Cockpit". But I can't install the "cockpit-dhcp" package. I could use third-party repositories, but apparently, using third-party repositories can with adverse consequences on the security and stability of my operating system.

So, I'm stuck on this problem, the only solution I see is to use an operating system adapted for "Cockpit", but my goal is to be able to create a DHCP server, a DNS server and a Samba server with "Cockpit" on Debian.

If you know of a solution that would enable me to achieve my goal, could you help me?

Thank you in advance for your reply.

**Comment by martinpitt** [comment, marked as answer]
There are a few [third-party extensions](https://cockpit-project.org/applications.html), but I'm not aware of any Cockpit UI for DHCP and DNS administration, sorry. Likewise, for a samba **client** there is [cockpit-file-sharing](https://github.com/45Drives/cockpit-file-sharing), but not for a Samba server. So I'm afraid there is no current solution for that, and you need to do these in the "classic" way with the command line, Ansible, etc.

**Reply by Angelo325** [reply]
Hello,

If I use Cockpit on an OS other than Debian (Fedora, Red Hat Enterprise Linux or Ubuntu), will I be able to create a DHCP, DNS or Samba server with Cockpit?

**Reply by martinpitt** [reply]
No -- there is no UI for that, this isn't specific to the underlying operating system.
